src/app/games/martian-chess/MartianChessDummyMinimax.ts

```diff
@@ -0,0 +1,171 @@
+import { MartianChessMove } from './MartianChessMove';
+import { MartianChessMoveResult, MartianChessNode, MartianChessRules } from './MartianChessRules';
+import { MartianChessPiece, MartianChessState } from './MartianChessState';
+
+export class MartianChessDummyMinimax extends Minimax<MartianChessMove, MartianChessState, MartianChessMoveResult> {
+
+    public constructor(public ruler: MartianChessRules, name: string) {
+        super(ruler, name);
+    }
+    public getListMoves(node: MartianChessNode): MartianChessMove[] {
+        const state: MartianChessState = node.gameState;
+        const currentPlayer: Player = state.getCurrentPlayer();
+        const yRange: [number, number] = state.getPlayerTerritory(currentPlayer);
+        const moves: MartianChessMove[] = [];
+        for (let y: number = yRange[0]; y <= yRange[1]; y++) {
+            for (let x: number = 0; x < 4; x++) {
+                const piece: MartianChessPiece = state.getPieceAtXY(x, y);
+                switch (piece) {
+                    case MartianChessPiece.PAWN:
+                        moves.push(...this.getMoveForPawn(x, y, state));
+                        break;
+                    case MartianChessPiece.DRONE:
+                        moves.push(...this.getMoveForDrone(x, y, state));
+                        break;
+                    case MartianChessPiece.QUEEN:
+                        moves.push(...this.getMoveForQueen(x, y, state));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+        return moves;
+    }
+    public getMoveForPawn(x: number, y: number, state: MartianChessState): MartianChessMove[] {
+        const coord: Coord = new Coord(x, y);
+        const landingCoords: Coord[] = [];
+        for (const dir of Direction.DIRECTIONS.filter((d: Direction) => d.isDiagonal())) {
+            const landingCoord: Coord = coord.getNext(dir);
+            if (landingCoord.isInRange(4, 8)) {
+                landingCoords.push(landingCoord);
+            }
+        }
+        return this.addMoves(state, coord, landingCoords);
+    }
+    private addMoves(state: MartianChessState,
+                     coord: Coord,
+                     landingCoords: Coord[])
+    {
+        const moves: MartianChessMove[] = [];
+        const firstPiece: MartianChessPiece = state.getPieceAt(coord);
+        const canCallTheClock: boolean = state.countDown.isAbsent();
+        const canPromoteDrone: boolean = state.isTherePieceOnPlayerSide(MartianChessPiece.DRONE) === false;
+        const canPromoteQueen: boolean = state.isTherePieceOnPlayerSide(MartianChessPiece.QUEEN) === false;
+        const last: MGPOptional<MartianChessMove> = state.lastMove;
+        for (const landingCoord of landingCoords) {
+            const landedPiece: MartianChessPiece = state.getPieceAt(landingCoord);
+            if (landedPiece === MartianChessPiece.EMPTY) {
+                // Moves
```

Comment:
"Regular moves"? Comment on appelle un coup « déplacement uniquement » en général ?

Reply:
displacement ? en français coup pour tous et déplacement/mouvement pour ceux sans capture.

Reply:
Displacement : non, c'est un terme très spécifique (« In geometry and mechanics, a displacement is a vector whose length is the shortest distance from the initial to the final position of a point P undergoing motion »). Satanés langages qui ont pas de termes clairs :D Donc j'ai pas l'impression qu'il y ait de terme adéquat, je te propose de résoudre et de laisser comme ça :(

Reply:
ou on nomme ça en ithkuil, on saura choisir précisément la spécificité qu'on veut et l'ambiguité qu'on veux dans nos mots !

Reply:
hmm :D Tu peux résoudre car c'est non résolvable je pense !

Reply:
bah je ne peux pas du coup !
lolelul, tu peux résoudre (si si)